core/src/test/scala/unit/kafka/zk/EmbeddedZookeeper.scala

```diff
@@ -48,6 +48,7 @@ class EmbeddedZookeeper() extends Logging {
 
   def shutdown() {
     CoreUtils.swallow(zookeeper.shutdown(), this)
+    CoreUtils.swallow(zookeeper.getZKDatabase().close(), this)
```

Comment:
Good find! Looking closer, it looks like `factory.shutdown()` calls `zkServer.shutdown()` so we don't need the `zookeeper.shutdown()` call. And perhaps `zookeeper.getZKDatabase().close()` should be after `Iterator.continually(isDown()).exists(identity)`?